gcloud/monitoring/group.py

```diff
@@ -0,0 +1,502 @@
+        self.filter = filter_string
+        self.is_cluster = is_cluster
+
+        if group_id:
+            self._name = group_name_from_id(client.project, group_id)
+        else:
+            self._name = None
+
+    @property
+    def id(self):
+        """Returns the group ID.
+
+        :rtype: str or None
+        :returns: the ID of the group based on it's name.
+        """
+        return self._id
+
+    @property
+    def name(self):
+        """Returns the fully qualified name of the group.
+
+        :rtype: str or None
+        :returns:
+            The fully qualified name of the group in the format
+            "projects/<project>/groups/<id>".
+        """
+        return self._name
+
+    @property
+    def parent_name(self):
+        """Returns the fully qualified name of the parent group.
+
+        :rtype: str or None
+        :returns:
+            The fully qualified name of the parent group.
+        """
+        if not self.parent_id:
+            return None
+        return group_name_from_id(self.client.project, self.parent_id)
+
+    @property
+    def path(self):
+        """URL path to this group.
+
+        :rtype: str
+        :returns: the path based on project and group name.
+
+        :raises: :exc:`ValueError` if :attr:`name` is not specified.
+        """
+        if not self.id:
+            raise ValueError('Cannot determine path without group ID.')
+        return '/' + self.name
+
+    def create(self):
+        """Create a new group based on this object via a ``POST`` request.
+
+        Example::
+
+            >>> group = client.group(
+            ...     display_name='My group',
+            ...     filter_string='resource.type = "gce_instance"',
+            ...     parent_id='5678',
+            ...     is_cluster=True)
+            >>> group.create()
+
+        The ``name`` attribute is ignored in preparing the creation request.
+        All attributes are overwritten by the values received in the response
+        (normally affecting only ``name``).
+        """
+        path = self.path_helper(self.client.project)
+        info = self.client.connection.api_request(
+            method='POST', path=path, data=self._to_dict())
+        self._init_from_dict(info)
+
+    def exists(self):
+        """Test for the existence of the group via a ``GET`` request.
+
+        :rtype: bool
+        :returns: Boolean indicating existence of the group.
+        """
+        try:
+            self.client.connection.api_request(method='GET', path=self.path)
+        except NotFound:
+            return False
+        else:
+            return True
+
+    def reload(self):
+        """Sync local group information via a ``GET`` request.
+
+        .. warning::
+
+            This will overwrite any local changes you've made and not saved
+            via :meth:`update`.
+        """
+        info = self.client.connection.api_request(method='GET', path=self.path)
+        self._init_from_dict(info)
```

Comment:
`init` is the wrong verb here, since the instance is being updated, not initialized